### PR TITLE
Add `Rootstock` mainnet chain

### PR DESCRIPTION
**Pull request by ahsan-javaiid**
### Description

This PR adds Rootstock main chain config to hyperlane registry. Previously testnet configurations merged https://github.com/hyperlane-xyz/hyperlane-registry/pull/115

### Contract Deployment

Hyperlane contract on Rootstock mainnet will be deployed using multisig account discussed between Rootstock and Hyperlane team. Later ownership could be transferred to someone in community builders on Rootstock and Hyperlane. 

## About Rootstock
Website: https://rootstock.io/
Defillama: https://defillama.com/chain/Rootstock
Explorer: https://explorer.rootstock.io/
Developer portal: https://dev.rootstock.io/

### Backward compatibility

Are these changes backward compatible? Note that additions are backwards compatible.

Yes

### Testing

Tested sending / receiving messages to / from rootstock chain.

### CODE REVIEW

**Comment by ahsan-javaiid** [comment]
cc: @ jmrossy for review.

File: chains/rootstock/logo.svg [hunk 1]
```diff
@@ -0,0 +1 @@
....36-19.74-11.02-19.73-34.49,.02-45.5,16.17-8.76,27.16-25.89,27.16-45.57,0-.4-.02-.79-.03-1.18l.02,.03c-.51-22.86,20.07-34.73,39.61-22.85h0c15.65,9.5,35.87,10.39,52.84,.58,24.77-14.31,33.25-46,18.94-70.77-14.31-24.77-46-33.25-70.77-18.94-16.96,9.8-26.27,27.73-25.88,46.02h0c.51,22.81-20,34.68-39.5,22.91-7.87-4.82-17.11-7.6-27.01-7.6s-19.14,2.78-27,7.59c-19.46,11.73-39.91-.12-39.39-22.88h0c.4-18.33-8.96-36.3-25.97-46.09-24.8-14.26-56.47-5.72-70.73,19.08-14.26,24.8-5.72,56.47,19.08,70.73,16.95,9.75,37.11,8.84,52.73-.63h0c19.49-11.83,40.02,.03,39.5,22.83l.02-.03c0,.4-.03,.8-.03,1.21,0,19.69,10.98,36.81,27.16,45.57,19.76,11.02,19.75,34.52-.02,45.52-.8,.43-1.58,.89-2.35,1.36-19.46,11.71-39.9-.15-39.38-22.91h0c.22-9.21-2.01-18.6-6.96-27.16-14.34-24.76-46.03-33.2-70.79-18.87C1.09,226.32-7.35,258.01,6.98,282.77c14.34,24.76,46.03,33.2,70.79,18.87,.32-.19,.63-.39,.95-.58l-.03,.04c19.51-11.84,40.04,.03,39.51,22.84l.02-.03c0,.42-.03,.85-.03,1.27,.03,28.61,23.24,51.78,51.85,51.75,28.61-.03,51.78-23.24,51.75-51.85,0-.38-.02-.76-.03-1.14l.02,.03c-.52-22.87,20.07-34.75,39.62-22.87l-.03-.05c.36,.22,.7,.44,1.06,.65,24.79,14.29,56.46,5.78,70.75-19.01,14.29-24.79,5.78-56.46-19.01-70.75Z"/></g></g></svg>
```
Comment (paulbalaji) [inline chains/rootstock/logo.svg:1]: ran through https://jakearchibald.github.io/svgomg/
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" id="Layer_2" viewBox="0 0 340.13 376.93"><defs><style>.cls-1{fill:#ff931e}</style></defs><g id="Layer_1-2"><circle cx="170.16" cy="51.8" r="51.8" class="cls-1"/><path d="M314.19 211.93c-24.79-14.29-56.46-5.78-70.75 19.01a51.582 51.582 0 0 0-6.92 27.04c.52 22.83-20.01 34.71-39.53 22.92-.77-.47-1.56-.93-2.36-1.36-19.74-11.02-19.73-34.49.02-45.5 16.17-8.76 27.16-25.89 27.16-45.57 0-.4-.02-.79-.03-1.18l.02.03c-.51-22.86 20.07-34.73 39.61-22.85 15.65 9.5 35.87 10.39 52.84.58 24.77-14.31 33.25-46 18.94-70.77-14.31-24.77-46-33.25-70.77-18.94-16.96 9.8-26.27 27.73-25.88 46.02.51 22.81-20 34.68-39.5 22.91-7.87-4.82-17.11-7.6-27.01-7.6s-19.14 2.78-27 7.59c-19.46 11.73-39.91-.12-39.39-22.88.4-18.33-8.96-36.3-25.97-46.09-24.8-14.26-56.47-5.72-70.73 19.08-14.26 24.8-5.72 56.47 19.08 70.73 16.95 9.75 37.11 8.84 52.73-.63 19.49-11.83 40.02.03 39.5 22.83l.02-.03c0 .4-.03.8-.03 1.21 0 19.69 10.98 36.81 27.16 45.57 19.76 11.02 19.75 34.52-.02 45.52-.8.43-1.58.89-2.35 1.36-19.46 11.71-39.9-.15-39.38-22.91.22-9.21-2.01-18.6-6.96-27.16-14.34-24.76-46.03-33.2-70.79-18.87-24.81 14.33-33.25 46.02-18.92 70.78 14.34 24.76 46.03 33.2 70.79 18.87.32-.19.63-.39.95-.58l-.03.04c19.51-11.84 40.04.03 39.51 22.84l.02-.03c0 .42-.03.85-.03 1.27.03 28.61 23.24 51.78 51.85 51.75 28.61-.03 51.78-23.24 51.75-51.85 0-.38-.02-.76-.03-1.14l.02.03c-.52-22.87 20.07-34.75 39.62-22.87l-.03-.05c.36.22.7.44 1.06.65 24.79 14.29 56.46 5.78 70.75-19.01 14.29-24.79 5.78-56.46-19.01-70.75Z" class="cls-1"/></g></svg>

```

File: chains/rootstock/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  confirmations: 1
+  estimateBlockTime: 30
+  reorgPeriod: 4
+chainId: 30
+deployer:
+  # Deployment in discussion between Rootstock <> Abacus Works.
+  name: Abacus Works
+  url: https://www.hyperlane.xyz
+displayName: Rootstock
+domainId: 30
```
Comment (paulbalaji) [inline chains/rootstock/metadata.yaml:17]: just one more thing to add
```suggestion
gasCurrencyCoinGeckoId: rootstock
```

File: chains/rootstock/metadata.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+nativeToken:
+  decimals: 18
+  name: Rootstock Smart Bitcoin
+  symbol: RBTC
+protocol: ethereum
+rpcUrls:
+  # Note: Using Rootstock RPC api here which contains APIKEY in url.
+  # Recommended way is to get a new APIKEY whoever wants to run Hyperlane on Rootstock.
+  # RPC API Dashboard: https://dashboard.rpc.rootstock.io
+  - http: https://rpc.mainnet.rootstock.io/kXhXHf6TnnfW1POvr4UT0YUvujmuju-M
```
Comment (paulbalaji) [inline chains/rootstock/metadata.yaml:29]: Adding some more fallback public RPCs
```suggestion
  - http: https://rpc.mainnet.rootstock.io/kXhXHf6TnnfW1POvr4UT0YUvujmuju-M
  - http: https://public-node.rsk.co
  - http: https://rootstock.drpc.org
  - http: https://mycrypto.rsk.co
```

File: chains/rootstock/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+transactionOverrides:
+  gasLimit: 6800000
```
Comment (paulbalaji) [inline chains/rootstock/metadata.yaml:31]: a link to dev docs would be great if possible!

File: chains/rootstock/metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  estimateBlockTime: 30
+  reorgPeriod: 4
+chainId: 30
+deployer:
+  # Deployment in discussion between Rootstock <> Abacus Works.
+  name: Abacus Works
+  url: https://www.hyperlane.xyz
+displayName: Rootstock
+domainId: 30
+isTestnet: false
```
Comment (paulbalaji) [inline chains/rootstock/metadata.yaml:18]: not required for mainnets
```suggestion
```

Comment (ahsan-javaiid) [reply chains/rootstock/metadata.yaml]: Fixed!

File: chains/rootstock/metadata.yaml (hunk 4 above)
Comment (ahsan-javaiid) [reply chains/rootstock/metadata.yaml:37]: Fixed!

File: chains/rootstock/metadata.yaml (hunk 3 above)
Comment (ahsan-javaiid) [reply chains/rootstock/metadata.yaml:30]: Fixed!

File: chains/rootstock/metadata.yaml (hunk 2 above)
Comment (ahsan-javaiid) [reply chains/rootstock/metadata.yaml:17]: Fixed!

File: chains/rootstock/logo.svg (hunk 1 above)
Comment (ahsan-javaiid) [reply chains/rootstock/logo.svg]: Fixed!

**Review by jmrossy — APPROVED** [review]

**Pull request merged**